cpp/src/com/google/idea/blaze/cpp/BlazeCWorkspace.java

```diff
@@ -313,13 +313,18 @@ private WorkspaceModel calculateConfigurations(
             .map(File::getAbsolutePath)
             .forEach(compilerSwitchesBuilder::withSystemIncludePath);
 
-        // add builtin includes provided by the compiler as system includes
-        // Note: In most cases CLion is able to derive the builtin includes during compiler info collection, unless
-        // the toolchain uses an external sysroot.
-        compilerSettings.getBuiltInIncludes().stream()
-            .flatMap(resolver)
-            .map(File::getAbsolutePath)
-            .forEach(compilerSwitchesBuilder::withSystemIncludePath);
+        // add includes from a custom sysroot as system includes
+        // Note: Only add includes from a custom sysroot manually, CLion can derive the other
+        // bulletin includes during the compiler info collection. Manually adding all builtin
+        // includes can lead to headers being resolved into the wrong include directory.
+        final var sysroot = compilerSettings.getSysroot();
+        if (sysroot != null) {
+          compilerSettings.getBuiltInIncludes().stream()
```

Comment:
let's revise this PR and consider doing this with registry key. I'm unsure if this change as is would lead to some unresolved headers etc, this has to be validated with custom toolchain with sysroot.